cico_setup.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+# Output command before executing
+set -x
+
+# Exit on error
+set -e
+
+# Source environment variables of the jenkins slave
+# that might interest this worker.
+function load_jenkins_vars() {
+  if [[ -e "jenkins-env" ]]; then
+    cat jenkins-env \
+      | grep -E "(DEVSHIFT_TAG_LEN|QUAY_USERNAME|QUAY_PASSWORD|JENKINS_URL|GIT_BRANCH|GIT_COMMIT|BUILD_NUMBER|ghprbSourceBranch|ghprbActualCommit|BUILD_URL|ghprbPullId)=" \
+      | sed 's/^/export /g' \
+      > ~/.jenkins-env
+    source ~/.jenkins-env
+  fi
+}
+
+function install_deps() {
+  # We need to disable selinux for now, XXX
+  /usr/sbin/setenforce 0 || :
+
+  # Get all the deps in
+  sudo yum -y install \
+    docker \
+    make \
+    git \
+    curl
+
+  service docker start
+
+  echo 'CICO: Dependencies installed'
+}
+
+function prepare() {
+  # Let's test
+  make docker-start
+  make -d docker-check-go-format
+  make -d docker-deps
+  make -d docker-check-go-code
+  make -d docker-build
+  echo 'CICO: Preparation complete'
+}
+
+function run_tests_without_coverage() {
+  make docker-test-unit
+  echo "CICO: ran tests without coverage"
+}
+
+function run_tests_with_coverage() {
+  # Run the unit tests that generate coverage information
+  make docker-test-unit-with-coverage
+
+  # Output coverage
+  make docker-coverage-all
+
+  # Upload coverage to codecov.io
+  cp tmp/coverage.mode* coverage.txt
+  bash <(curl -s https://codecov.io/bash) -X search -f coverage.txt -t 6539a1b6-70f9-4ba1-8ab5-d1c8a104ec83 #-X fix
```

Comment:
@tinakurian when you continue to use this token you will be pushing coverage information to the toolchain-operator project because that uses the same hash (https://github.com/fabric8-services/toolchain-operator/blob/83f2e947ad81bcdf30beb507c5cb589ac9c74ded/cico_setup.sh#L62).

Reply:
Thanks Konrad, addressed here: https://github.com/redhat-developer/devopsconsole-operator/pull/17